### PR TITLE
fetched data

### PR DESCRIPTION
**Pull request by silchyn**
[DEMO](https://silchyn.github.io/js_fetch_data_DOM/)

### CODE REVIEW

File: src/scripts/main.js [hunk 1]
```diff
@@ -1,3 +1,56 @@
...
-// write your code here
+function request(url) {
+  return fetch(url)
+    .then(response => Promise.resolve(response.json()))
+    .catch(error => Promise.reject(error));
+}
+
+function getPhones() {
+  return request(
+    'https://mate-academy.github.io/phone-catalogue-static/api/phones.json');
```
Comment (Ihor-Karpyn) [inline src/scripts/main.js:11]: Create variables `BASE_URL` and `ENDPOINTS = {... }`

File: src/scripts/main.js [hunk 2]
```diff
@@ -1,3 +1,56 @@
...
+  .then(phones => {
+    showNames(phones);
+
+    getPhonesDetails(getIds(phones))
+      .then(response => {
+        const phonesWithDetails = [];
+
+        for (let i = 0; i < phones.length; ++i) {
+          phonesWithDetails.push(Object.assign(phones[i], response[i]));
+        }
```
Comment (Ihor-Karpyn) [inline src/scripts/main.js:54]: use array iteration methods, forEach or reduce

**Review by Ihor-Karpyn — CHANGES_REQUESTED** [review]

File: src/scripts/main.js [hunk 3]
```diff
@@ -1,3 +1,54 @@
...
+}
+
+function getIds(phones) {
+  const result = [];
+
+  for (const phone of phones) {
+    result.push(phone.id);
+  }
+
+  return result;
```
Comment (Ihor-Karpyn) [inline src/scripts/main.js:25]: reduce

File: src/scripts/main.js [hunk 4]
```diff
@@ -1,3 +1,54 @@
...
+
+function request(url) {
+  return fetch(url)
+    .then(response => Promise.resolve(response.json()));
+}
+
+function getPhones() {
+  return request(`${BASE_URL}${ENDPOINTS.phones}`);
+}
+
```
Comment (Ihor-Karpyn) [inline src/scripts/main.js:17]: Better to move in different file like API

File: src/scripts/main.js [hunk 5]
```diff
@@ -1,3 +1,54 @@
...
+getPhones()
+  .then(phones => {
+    showNames(phones);
+
+    getPhonesDetails(getIds(phones))
+      .then(response => {
+      // const phonesWithDetails = phones.reduce((accumulator, current, i) => {
+        //   return [...accumulator, Object.assign(current, response[i])];
+        // }, []);
```
Comment (Ihor-Karpyn) [inline src/scripts/main.js:52]: ?

File: src/scripts/main.js [hunk 6]
```diff
@@ -1,3 +1,54 @@
...
+  }
+
+  return result;
+}
+
+function getPhonesDetails(ids) {
+  return Promise.all(ids.map(id => {
+    return request(`${BASE_URL}${ENDPOINTS.phoneById(id)}`);
+  }));
+}
```
Comment (Ihor-Karpyn) [inline src/scripts/main.js:32]: and this one

**Review by Ihor-Karpyn — CHANGES_REQUESTED** [review]

File: src/scripts/main.js (hunk 5 above)
Comment (silchyn) [reply src/scripts/main.js:52]: Linter swears at a variable that is not used, but the task says that it needs to be implemented.

Comment (Ihor-Karpyn) [reply src/scripts/main.js:52]: got it

**Review by Ihor-Karpyn — APPROVED** [review]

**Pull request closed without merging**